compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -352,6 +353,31 @@ tests tools@Tools{damlc} = testGroup "Packaging" $
             ]
         buildProjectError projDir "" "name collision"
 
+    , testCase "Virtual module name collision" $ withTempDir $ \projDir -> do
+        createDirectoryIfMissing True (projDir </> "src" </> "A" </> "B")
+        writeFileUTF8 (projDir </> "daml.yaml") $ unlines
+            [ "sdk-version: " <> sdkVersion
+            , "name: proj"
+            , "version: 0.0.1"
+            , "source: src"
+            , "dependencies: [daml-prim, daml-stdlib]"
+            ]
+        writeFileUTF8 (projDir </> "src" </> "A.daml") $ unlines
+            [ "module A where"
+            , "data B = B Int"
+            ]
+        writeFileUTF8 (projDir </> "src" </> "A" </> "B" </> "C.daml") $ unlines
+            [ "module A.B.C where"
+            , "data C = C Int"
+            ]
+        (exitCode, out, err) <- readProcessWithExitCode damlc ["build", "--project-root", projDir] ""
+        out @?= unlines
+            [ "Compiling proj to a DAR."
+            , "Created .daml/dist/proj-0.0.1.dar"
+            ]
+        assertInfixOf "collision between variant A:B and virtual module A.B (from A.B.C)" err
```

Comment:
What do you think of speaking about "module name prefixes" in the error messages?

Reply:
Changed it to use `module prefix` instead of `virtual module` in the error message. I still call it virtual module internally since I think that makes the code easier to understand since it treats them pretty much exactly like other module names.

Reply:
I totally agree regarding the internal naming.